src/main/java/org/terasology/fluid/system/FluidAuthoritySystem.java

```diff
@@ -60,6 +70,10 @@ public void fillFluidContainerItem(ActivateEvent event, EntityRef item, FluidCon
 
                         FluidUtils.setFluidForContainerItem(removedItem, fluidType);
 
+                        // Fill this fluid container up to max capacity.
```

Comment:
This should probably live in the FluidUtils.setFluidForContainerItem(...) method so that it gets saved along with the fluid type change.